Dockerfile

```diff
@@ -1,18 +1,59 @@
-FROM ubuntu:xenial
+# syntax=docker/dockerfile:experimental
 
-RUN apt-get update && apt-get install --yes nginx
+# Build stage: Install ruby dependencies
+# ===
+FROM ruby:2.6-slim AS ruby-dependencies
+RUN apt-get update && apt-get install --no-install-recommends -y build-essential ruby-dev ruby-bundler
```

Comment:
Doesn't the `ruby` image come with `gem` and `bundler`?

Reply:
I don't think this works. You no longer seem to be building jekyll, and you are still importing `ruby-dev` and `ruby-bundler`.

I'd suggest that this might work:

``` Dockerfile
FROM ruby:2.6-slim AS jekyll-build
WORKDIR /srv
ADD . .
RUN bundle install
RUN bundle exec jekyll build
```

Reply:
That fails with bundle install

Reply:
Okay, I just downloaded it and tried it. Yes it fails because of [an FFI issue](https://stackoverflow.com/questions/7852566/error-error-installing-ffi-error-failed-to-build-gem-native-extension). We could look into a few ways around that.

Reply:
I believe this works if you use `ruby:2.6` instead of `ruby:2.6-slim`. The question is, is that better or worse than using `2.6-slim` or `ubuntu:bionic` and then installing `ruby-dev`.